app/models/menu.rb

```diff
@@ -0,0 +1,18 @@
+class Menu < ApplicationRecord
+  has_many :menu_items, dependent: :destroy
+  has_many :orders
+
+  accepts_nested_attributes_for :menu_items, allow_destroy: true, reject_if: proc { |att| att['name'].blank? }
```

Comment:
`reject_if: proc { |att| att['name'].blank? }` will create you problems in the future. Try to depend mostly on validations, they are more talkative